spring-boot-project/spring-boot-test/src/main/java/org/springframework/boot/test/context/SpringBootTestContextBootstrapper.java

```diff
@@ -135,7 +135,7 @@ protected ContextLoader resolveContextLoader(Class<?> testClass,
 	}
 
 	private void addConfigAttributesClasses(ContextConfigurationAttributes configAttributes, Class<?>[] classes) {
-		List<Class<?>> combined = new ArrayList<>(Arrays.asList(classes));
+		Set<Class<?>> combined = new HashSet<>(Arrays.asList(classes));
```

Comment:
I wonder if this should be a `LinkedHashSet` or similar? Assuming that the ordering is preserved elsewhere, losing it here by switching to a `HashSet` could, in theory, have an effect on the beans that are defined as it'll change the order in which they're processed during refresh.

Reply:
Good point - I'll change this.